tests/integration_test.go

```diff
@@ -0,0 +1,90 @@
+package tests
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"testing"
+
+	skynet "github.com/NebulousLabs/go-skynet"
+	"gopkg.in/h2non/gock.v1"
+)
+
+// TestUploadAndDownloadFile tests uploading and downloading a single file.
+func TestUploadAndDownloadFile(t *testing.T) {
+	defer gock.Off() // Flush pending mocks after test execution
+
+	const srcFile = "../testdata/file1.txt"
+	const skylink = skynet.URISkynetPrefix + "testskynet"
+
+	// Test upload.
+
+	gock.New(skynet.DefaultUploadOptions.PortalURL).
+		Post(skynet.DefaultUploadOptions.PortalUploadPath).
+		Reply(200).
+		JSON(map[string]string{"skylink": "testskynet"})
+
+	skylink2, err := skynet.UploadFile(srcFile, skynet.DefaultUploadOptions)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if skylink2 != skylink {
+		t.Fatalf("expected skylink %v, got %v", skylink, skylink2)
+	}
+
+	// Test download.
+
+	gock.New(skynet.DefaultDownloadOptions.PortalURL).
+		Get(skynet.DefaultDownloadOptions.PortalDownloadPath).
+		Reply(200).
+		BodyString("test\n")
+
+	file, err := ioutil.TempFile("", t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	dstFile := file.Name()
+
+	err = skynet.DownloadFile(dstFile, skylink, skynet.DefaultDownloadOptions)
```

Comment:
This should verify the behaviour of `DownloadFile`. For example here it should verify the prefix is being trimmed from the given skylink when the request is being formed.

```
	resp, err := http.Get(fmt.Sprintf("%s/%s", strings.TrimRight(opts.PortalURL, "/"), strings.TrimPrefix(skylink, "sia://")))
	if err != nil {
```

we should probably also try and mock error responses and cover how the sdk handles those